### PR TITLE
CreateIfNotExists throws when TableBeingDeleted

### PR DESCRIPTION
**Pull request by christothes**
#23600

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ namespace Azure.Data.Tables
     public class TableClient
     {
         private static readonly char[] ContinuationTokenSplit = { ' ' };
-        private readonly ClientDiagnostics _diagnostics;
+        internal readonly ClientDiagnostics _diagnostics;
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:25]: revert?

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs]: Did it just before this comment showed up :)

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 2]
```diff
@@ -438,7 +438,7 @@ public virtual Response<TableItem> CreateIfNotExists(CancellationToken cancellat
                     cancellationToken);
                 return Response.FromValue(response.Value as TableItem, response.GetRawResponse());
             }
-            catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.Conflict)
+            catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.Conflict && ex.ErrorCode != TableErrorCode.TableBeingDeleted)
             {
                 return default;
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:443]: Wait, are we returning null from this method? It's against the guidelines...

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:443]: yes - This must have been missed before.

**Review by pakrym — APPROVED** [review]
this particular change LGTM but we need to fix the null-returning branches.

**Pull request merged**